bitstreams.md

```diff
@@ -37,14 +37,87 @@ Exposed links:
 
 Bitstream metadata can be modified as described in [Modifying metadata via Patch](metadata-patch.md).
 
+Additional bitstream properties can be modified via Patch as described below.
+
+### Replace
+The replace operation allows to replace *existent* information with new one. Attempt to use the replace operation to set not yet initialized information must return an error. See [general errors on PATCH requests](patch.md)
+
+To change the bundle of a bitstream:
+
+`curl --data '[ { "op": "replace", "path": "/bundleName", "value": "TEST"}]' -H "Authorization: Bearer ..." -H "content-type: application/json" -X PATCH ${dspace7-url}/api/core/bitstreams/${uuid}`
+
+For example, starting with the following bitstream data:
+```json
+{
+  "name" : "test.zip",
+  "bundleName" : "ORIGINAL",
+  "sequenceId" : 5
+}
+```
+the change bundleName operation will result in:
+```json
+{
+  "name" : "test.zip",
+  "bundleName" : "TEST",
+  "sequenceId" : 5
+}
+```
+
+To change the sequenceId:
```

Comment:
which is the use case for this change? my understanding is that the sequenceId is only managed internally by dspace itself to produce a pseudo-persistent URL of the bitstream, so we should not allow to change it
https://wiki.duraspace.org/display/DSDOC5x/Functional+Overview

Reply:
This is currently supported in DSpace 6: https://github.com/DSpace/DSpace/blob/dspace-6_x/dspace-rest/src/main/java/org/dspace/rest/BitstreamResource.java#L474

Reply:
hi @benbosman thanks to point out that the "old" REST allows that but do you know if this make really sense for the users? I just asking because I'm not able to identify the business need / opportunity to allow that here and we should not reintroduce an "error" if so just because it was allowed in the old (not widely used) REST

Reply:
IIRC (I could be mistaken), isn't the sequenceId necessary to update from the UI when you are *reordering bitstreams*? However, I guess reordering might be achievable in a different way in the REST API (e.g. by sending a reordered text/uri-list).

I agree though that we might want to approach this from the standpoint of use cases.  What is the sequenceID used for (primarily) and is there a better way of achieving that through the REST API *without* requiring the client to understand or even use sequenceID?

Reply:
I think in DSpace 6, the bitstream order is only based on the bundle2bitstream's bitstream_order
There's still config for this, but I don't see any relevant implementation in DSpace 6 or 7:  https://github.com/DSpace/DSpace/blob/master/dspace/config/dspace.cfg#L1566 vs https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/content/Bundle.java#L59

The sequence ID is used as the download ID in DSpace 6. But in DSpace 7, I'm starting to wonder whether the sequence ID should rather be removed completely.
Would any of you know a feature where the sequence ID is still relevant in DSpace 7?

Reply:
@benbosman : Thanks for the correction around bistream_order vs sequence_id,  I definitely mixed them up

I agree with you, I'm not sure I understand why/how the sequence ID is still necessary in DSpace 7, except maybe as a way to still redirect old URLs to the proper bitstream.

In DSpace 6.x, the sequence ID was used as an alternative way of locating a Bitstream...either the Bitstream Name or Sequence ID could be used (or both).  Here's an example where a Bitstream is downloaded from the XMLUI: https://github.com/DSpace/DSpace/blob/dspace-6_x/dspace-xmlui/src/main/java/org/dspace/app/xmlui/cocoon/BitstreamReader.java#L242-L249

So, for DSpace 7, I imagine we may need to *read* the existing "sequence id" to ensure that old Bitstream URLs can be redirected to new Bitstream URLs (as they may not be identical).  However, maybe we should consider making this property read-only (and documenting it as such)?  We could even deprecate it with the idea of removing it in DSpace 8, unless anyone sees a reason to keep it around?

Reply:
@tdonohue This makes sense to me as well.
Making sure it's deprecated will also ensure no further confusing will occur why the sequence_id doesn't determine the sequence :)

Reply:
As decided in today's meeting, the sequence_id will be READ-ONLY for now (in the REST API). But, we will not deprecate it yet...instead I'll create a JIRA ticket for discussion of whether to deprecate & remove (or not).   Here's today's discussion notes: https://wiki.duraspace.org/display/DSPACE/2019-08-22+DSpace+7+Working+Group+Meeting#id-2019-08-22DSpace7WorkingGroupMeeting-Notes